dockerfiles/base/scripts/base/startup.sh

```
@@ -251,20 +255,40 @@ init() {
   source "${SCRIPTS_CONTAINER_SOURCE_DIR}"/cli.sh
 }
 
+cleanup() {
+  RETURN_CODE=$?
+
+  # CLI developers should only return '3' in code after the init() method has completed.
+  # This will check to see if the CLI directory is not mounted and only offer the error
+  # message if it isn't currently mounted.
+  if [ $RETURN_CODE -eq "3" ]; then
```

As far as I understand we should check that no code is using exit or return with exit code that differs from 3 in that case. Am I right? Have you grepped CLI sources such a code?

---

In the current CLI implementation, there are no methods that return a '3'. So, this is laying a groundwork so that as we improve the CLI over time, if there are CLI crashes or hard error messages that we print out, we should return this error code to trigger this help message.

---

The other thing we could do, @riuvshin - is not even bother with a :/cli volume mount. Instead, we could always have the cli.log written to :/data volume mount which is the root mount that is always required.

---

I would like yes that in case of error, error log is always dumped in /data so we're sure to have the dump at the right time. Re-running cli with a /cli mount may not reproduce the issue (and then we don't have the log)

---

+1

---

Fixed.